### PR TITLE
[MTDSA-4418] add Submit Property Adjustments controller, route, IT tests

### PR DESCRIPTION
**Pull request by MarkAKelly**

### CODE REVIEW

File: app/v1/models/errors/mtdErrors.scala [hunk 1]
```diff
@@ -38,34 +44,41 @@ object TypeOfBusinessFormatError
 object AdjustedStatusFormatError
   extends MtdError("FORMAT_ADJUSTED_STATUS", "The supplied adjusted status format is invalid")
 
+object FormatAdjustmentValueError extends MtdError("FORMAT_ADJUSTMENT_VALUE", "") {
+  def withFieldName(fieldName: String): MtdError = this.copy(message = s"The format of the '$fieldName' value is invalid")
```
Comment (BetaDraconis) [inline app/v1/models/errors/mtdErrors.scala]: This doesn't match the message in the requirement spec "The format of [fieldName] value is invalid"

File: app/v1/models/errors/mtdErrors.scala [hunk 2]
```diff
@@ -38,34 +44,41 @@ object TypeOfBusinessFormatError
...
+}
+
 object SelfEmploymentIdFormatError extends MtdError("FORMAT_SELF_EMPLOYMENT_ID","The supplied self-employment ID format is invalid")
 
+object BsasIdFormatError extends  MtdError("FORMAT_BSAS_ID", "The provided Business Source Adjustable Summary ID format is invalid")
```
Comment (BetaDraconis) [inline app/v1/models/errors/mtdErrors.scala]: Message should be "The format of the BSAS ID is invalid"

File: app/v1/models/errors/mtdErrors.scala [hunk 3]
```diff
@@ -38,34 +44,41 @@ object TypeOfBusinessFormatError
...
-object BothExpensesError
-  extends MtdError("RULE_BOTH_EXPENSES_SUPPLIED", "Both Expenses and consolidated expenses cannot be present at the same time")
-
-object SelfEmploymentAdjusted
-  extends MtdError("RULE_SELF_EMPLOYMENT_ADJUSTED", "A self-employment business type was adjusted. Re-trigger an adjustable summary for the self-employment to correct")
+object RuleBothExpensesError
+  extends MtdError("RULE_BOTH_EXPENSES_SUPPLIED", "Both expenses and consolidated expenses cannot be present at the same time")
 
-object BsasIdFormatError extends  MtdError("FORMAT_BSAS_ID", "The provided Business Source Accounting Summary ID format is invalid")
+object RuleSelfEmploymentAdjusted
```
Comment (BetaDraconis) [inline app/v1/models/errors/mtdErrors.scala]:
```suggestion
object RuleSelfEmploymentAdjustedError
```

File: app/v1/models/errors/mtdErrors.scala (hunk 1 above)
Comment (MarkAKelly) [reply app/v1/models/errors/mtdErrors.scala]: It does now.

File: app/v1/models/errors/mtdErrors.scala (hunk 2 above)
Comment (MarkAKelly) [reply app/v1/models/errors/mtdErrors.scala]: I had changed this but it probably snuck back in on the rebase

File: app/v1/models/errors/mtdErrors.scala (hunk 3 above)
Comment (MarkAKelly) [reply app/v1/models/errors/mtdErrors.scala]: We need to come up with consistent naming convention for these errors.  Half of the package is suffixed with error and half isn't

File: test/v1/fixtures/SubmitUKPropertyBsasRequestBodyFixtures.scala [hunk 4]
```diff
@@ -324,4 +341,137 @@ object SubmitUKPropertyBsasRequestBodyFixtures {
       )
     )
   }
+
+  val desResponse: String => String = (bsasId: String) =>
```
Comment (BetaDraconis) [inline test/v1/fixtures/SubmitUKPropertyBsasRequestBodyFixtures.scala]: might be worth adding 'fhl' to the val name to make it clear what type of property it belongs to.

File: test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+             |  "rel": "self"
+             | },
+             | {
+             |  "href": "/individuals/self-assessment/adjustable-summary/$nino/property/$bsasId?adjustedStatus=true",
+             |  "method": "GET",
+             |  "rel": "retrieve-adjustable-summary"
+             | }
+             | ]
+             |}
+             |""".stripMargin)
```
Comment (BetaDraconis) [inline test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala]: Could you not just use the method from the controllerISpec here?

File: test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
```
Comment (BetaDraconis) [inline test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala]: See above

File: test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+      }
+
+      val input = Seq(
+        (BadRequestError, BAD_REQUEST),
+        (NinoFormatError, BAD_REQUEST),
+        (RuleIncorrectOrEmptyBodyError, BAD_REQUEST),
+        (DownstreamError, INTERNAL_SERVER_ERROR),
+        (RuleBothExpensesError, BAD_REQUEST),
+        (FormatAdjustmentValueError, BAD_REQUEST),
+        (RuleAdjustmentRangeInvalid, BAD_REQUEST)
```
Comment (BetaDraconis) [inline test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala]: BsasIdFormatError is missing

File: test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+          val result: Future[Result] = controller.submitUkPropertyBsas(nino, bsasId)(fakePostRequest(validfhlInputJson))
+
+          status(result) shouldBe expectedStatus
+          contentAsJson(result) shouldBe Json.toJson(mtdError)
+          header("X-CorrelationId", result) shouldBe Some(correlationId)
+        }
+      }
+
+      val input = Seq(
+        (NinoFormatError, BAD_REQUEST),
```
Comment (BetaDraconis) [inline test/v1/controllers/SubmitUkPropertyBsasControllerSpec.scala]: BsasIdFormatError missing here too

**Pull request merged**